### ISSUE TITLE
Overwriting modules - composer update

### ISSUE THREAD

**Issue by LukaszNiziolDWG** [AsgardCms/Platform#660]
Hey!

I made a few changes in the basic AsgardCMS modules. When I do 'composer update' my changes are overwritten.

I tried to remove entries responsible for the modules in 'composer.json' - in such a situation when the 'composer update' is executed,the modules are removed.

How can I protect modules from overwriting / deleting?

### System Details
- `PHP Version`: >5.6.4
- `Laravel Version`: 5.4.*
- `Asgardcms/Platform version`: 2.6.1

**Comment by LukaszNiziolDWG** [comment]
`{
    "name": "asgardcms/platform",
    "description": "The AsgardCms application.",
    "keywords": [
        "cms",
        "asgardcms",
        "multilingual",
        "laravel",
        "laravel5"
    ],
    "license": "MIT",
    "type": "project",
    "require": {
        "php": ">=5.6.4",
        "laravel/framework": "5.4.*",
        "nwidart/laravel-modules": "~1.0",
        "cartalyst/sentinel": "~2.0",
        "laravel/tinker": "~1.0",
        "spatie/laravel-responsecache": "v2"
    },
    "require-dev": {
        "fzaninotto/faker": "~1.4",
        "mockery/mockery": "0.9.*",
        "phpunit/phpunit": "~5.7",
        "symfony/css-selector": "3.1.*",
        "symfony/dom-crawler": "3.1.*",
        "barryvdh/laravel-debugbar": "~2.3",
        "orchestra/testbench": "^3.4",
        "league/flysystem-aws-s3-v3": "^1.0",
        "league/commonmark": "^0.15.4"
    },
    "autoload": {
        "classmap": [
            "database"
        ],
        "psr-4": {
            "App\\": "app/",
            "Modules\\": "Modules/"
        }
    },
    "autoload-dev": {
        "classmap": [
            "tests/TestCase.php"
        ]
    },
    "scripts": {
        "post-install-cmd": [
            "Illuminate\\Foundation\\ComposerScripts::postInstall",
            "php artisan optimize"
        ],
        "post-update-cmd": [
            "Illuminate\\Foundation\\ComposerScripts::postUpdate",
            "php artisan optimize"
        ]
    },
    "config": {
        "preferred-install": "dist"
    },
    "extra": {
        "branch-alias": {
            "dev-master": "2.1.x-dev"
        }
    },
    "minimum-stability": "dev",
    "prefer-stable": true
}`

**Comment by nWidart** [comment]
You can't, that's how composer works. You need to remove them from composer and then using git, or just manual copy/pasting, getting them back into your project.

**Comment by LukaszNiziolDWG** [comment]
I'm sorry, I can't quite understand.
I remove them from the composer by deleting entries from the 'require' section.
What do you mean by 'and then using git'?

**Comment by nWidart** [comment]
Git can be used to recover files. So I meant to use git to recover the deleted files by composer.

**Comment by ChristianGiupponi** [comment]
If you do not use GIT, and I strongly recommend to use it, just follow this steps:

- Remove the module from composer.json
- Copy the module from your project to another folder
- run `composer update` to remove the module
- Re-copy the module from the other folder to your project

Fiy: https://git-scm.com/

**Comment by LukaszNiziolDWG** [comment]
I use git and the repository.
The problem is that when another programmer downloads a project and performs 'composer update' it loses my changes. Is it really impossible to get around it in some way?

**Comment by nWidart** [comment]
No, we explained to you the steps you need to do to overcome the limitation.

**Comment by LukaszNiziolDWG** [comment]
In the case of such steps, errors arise. I have overwritten 'Core' and 'Page' fashions. When I temporarily remove the 'Core' module and execute the 'composer update' command, I get errors.
What should another developer download my project do? I don't understand why my private modules can remain, and basic (with asgardcms) overwrite and / or delete (after execute 'composer update' or 'composer install').

**Comment by nWidart** [comment]
Oh I missed your reply from an hour ago.

You don't need to delete the folder before composer... what you need is to restore it after composer has deleted it.

**Comment by Kodok-Jr** [comment]
so, how can i deleted several module from a core module. 
ex: Tag and Page module
